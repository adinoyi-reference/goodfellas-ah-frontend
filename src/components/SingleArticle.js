import React from 'react';
import Comment from './shared/Comment';
import '../styles/styles.scss';
import star from '../assets/star.png';
import bookmark from '../assets/bookmark.png';
import share from '../assets/share.png';
import dot from '../assets/dot-dot.png';
import love from '../assets/love.png';

const SingleArticle = (props) => {
  const { article } = props;
  const {
    read_time, user, title, image, body,
  } = article;
  const number = read_time.split(' ')[0];
  const minute = read_time.split(' ')[1];
  return (
    <div>
      <div className="single-page">
        <div className="article-header">
          <div className="author-icon">
            <img src={user.profile.image} className="author-image" alt="" />
            <div className="number">{number}</div>
            <div className="minutes">{minute}</div>
          </div>
          <div className="title-author">
            <div className="article-title">{title}</div>
            <div className="author-name">
              <div className="name">{`${user.firstname} ${user.lastname}`}</div>
              <button type="button">Follow</button>
            </div>
          </div>
        </div>
        <div className="article-container">
          <img className="article-image" src={image} alt="" />
          <div className="side-button">
            <img className="star" src={star} alt="" />
            <img className="bookmark" src={bookmark} alt="" />
            <img className="share" src={share} alt="" />
            <img className="dot-dot" src={dot} alt="" />
            <div className="like-count"><p>4</p></div>
            <img className="love" src={love} alt="" />
          </div>
          <div className="article-content">
            <p>{body}</p>

          </div>

        </div>

      </div>
      <hr className="article-line" />
      <Comment />
    </div>

  );
};

export default SingleArticle;
