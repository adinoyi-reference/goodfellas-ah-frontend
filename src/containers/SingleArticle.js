import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getAnArticle } from '../actions/articleActions';
import ArticleHeader from '../components/article/ArticleHeader';
import ArticleBody from '../components/article/ArticleBody';
import Comment from '../components/article/Comment';

export class Article extends Component {
  componentWillMount = () => {
    const { match, getAnArticle: getArticle } = this.props;
    const { slug } = match.params;

    if (slug) {
      getArticle(slug);
    }
  }

  singleArticle() {
    const { article } = this.props;
    return (
      <div>
        <div className="single-page">
          <ArticleHeader article={article} />
          <ArticleBody article={article} />
          <Comment />
        </div>
      </div>

    );
  }

  renderArticle() {
    const { article, error } = this.props;
    if (article) {
      return this.singleArticle();
    }
    return <div>{error}</div>;
  }


  render() {
    return (
      <div>
        { this.renderArticle() }
      </div>
    );
  }
}


const mapStateToProps = (state) => ({
  error: state.articles.articleError,
  article: state.articles.article,
});

export default connect(mapStateToProps, { getAnArticle })(Article);
