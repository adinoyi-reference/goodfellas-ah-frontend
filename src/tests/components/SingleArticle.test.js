import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import SingleArticle from '../../components/SingleArticle';

let wrapped;

const props = {
  article: { read_time: 1 },
};

beforeEach(() => {
  wrapped = shallow(<SingleArticle {...props} />);
});

describe('Signin UI', () => {
  describe('render features', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
