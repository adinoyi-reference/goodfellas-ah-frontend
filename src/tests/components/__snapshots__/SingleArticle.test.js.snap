// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signin UI render features container should render as expected 1`] = `
<div>
  <div
    className="single-page"
  >
    <div
      className="article-header"
    >
      <div
        className="author-icon"
      >
        <img
          alt=""
          className="author-image"
          src="author-icon.jpg"
        />
        <div
          className="number"
        >
          1
        </div>
        <div
          className="minutes"
        >
          minutes
        </div>
      </div>
      <div
        className="title-author"
      >
        <div
          className="article-title"
        />
        <div
          className="author-name"
        >
          <div
            className="name"
          >
            Afeez Awoyemi
          </div>
          <button
            type="button"
          >
            Follow
          </button>
        </div>
      </div>
    </div>
    <div
      className="article-container"
    >
      <img
        alt=""
        className="article-image"
      />
      <div
        className="side-button"
      >
        <img
          alt=""
          className="star"
          src="star.png"
        />
        <img
          alt=""
          className="bookmark"
          src="bookmark.png"
        />
        <img
          alt=""
          className="share"
          src="share.png"
        />
        <img
          alt=""
          className="dot-dot"
          src="dot-dot.png"
        />
        <div
          className="like-count"
        >
          <p>
            4
          </p>
        </div>
        <img
          alt=""
          className="love"
          src="love.png"
        />
      </div>
      <div
        className="article-content"
      >
        <p />
      </div>
    </div>
  </div>
  <hr
    className="article-line"
  />
  <div
    className="comments"
  >
    <div
      className="add-comment"
    >
      <img
        alt=""
        src="author-icon.jpg"
      />
      <input
        placeholder="write a comment....."
        type="text"
      />
    </div>
    <div
      className="each-comment"
    >
      <div
        className="comment-heading"
      >
        <img
          alt=""
          src="author-icon.jpg"
        />
        <div
          className="comment-name"
        >
          Afeez Awoyemi
        </div>
        <div
          className="time-chat"
        >
          <div>
            <img
              alt=""
              className="comment-number"
              src="comment.png"
            />
            <p>
              2
            </p>
          </div>
          <div>
            <img
              alt=""
              className="read-time"
              src="clock.png"
            />
            <p>
              5
            </p>
          </div>
        </div>
      </div>
      <div
        className="comment-body"
      >
        You’re drafting an open letter. You have to put something up on the best platform in Africa before the queen asked you to stop and not do it.
      </div>
      <div
        className="each-reply"
      >
        <div
          className="reply-heading"
        >
          <img
            alt=""
            src="author-icon.jpg"
          />
          <div
            className="reply-name"
          >
            Afeez Awoyemi
          </div>
          <div
            className="time-chat"
          >
            <div>
              <img
                alt=""
                className="comment-number"
                src="comment.png"
              />
              <p>
                2
              </p>
            </div>
            <div>
              <img
                alt=""
                className="read-time"
                src="clock.png"
              />
              <p>
                5
              </p>
            </div>
          </div>
        </div>
        <div
          className="reply-body"
        >
          You’re drafting an open letter. You have to put something up on the best platform in Africa before the queen asked you to stop and not do it.
        </div>
      </div>
    </div>
  </div>
</div>
`;
